### ISSUE TITLE
wasm-ld: support soname option

### ISSUE THREAD

**Issue by anuraaga** [llvm/llvm-project#62931]
When using LLVM with wasi-sdk to compile existing C projects, I find it is very common to run into a snag that they generally set `-soname` for the executable name, which `wasm-ld` does not support and errors with `unknown argument`. Would it be possible to add support for this flag to `wasm-ld` to make it easier to compile existing projects to WebAssembly?

**Comment by dschuff** [comment]
We've been discussing converging Emscripten's support for dylibs with wasi-sdk (including maybe having some equivalent to soname) so it would probably make sense to fix this along with that effort.
/cc @sbc100

**Comment by sbc100** [comment]
Yes, I would prefer not to ignore it, but to do something useful with it.

One option is to use it to set the module name in the name section?  Another is to expand on the existing `.dylink` section to include it.
